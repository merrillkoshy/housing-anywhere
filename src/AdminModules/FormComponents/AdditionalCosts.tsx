import React from "react";
import { Form, Col, Row } from "react-bootstrap";
import Switch from "react-switch";

const AdditionalCosts = ({
	water,
	setWater,
	gas,
	setGas,
	electricity,
	setElectricity,
	internet,
	setInternet,
}: {
	water: boolean;
	setWater: React.Dispatch<React.SetStateAction<boolean>>;
	gas: boolean;
	setGas: React.Dispatch<React.SetStateAction<boolean>>;
	electricity: boolean;
	setElectricity: React.Dispatch<React.SetStateAction<boolean>>;
	internet: boolean;
	setInternet: React.Dispatch<React.SetStateAction<boolean>>;
}) => {
	return (
		<>
			<header>Additional Costs</header>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Water
				</Form.Label>
				<Col>
					<Switch
						id="water"
						onChange={() => {
							setWater(!water);
						}}
						checked={water}
					/>
				</Col>
			</Form.Group>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Gas
				</Form.Label>
				<Col>
					<Switch
						id="gas"
						onChange={() => {
							setGas(!gas);
						}}
						checked={gas}
					/>
				</Col>
			</Form.Group>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Electricity
				</Form.Label>
				<Col>
					<Switch
						id="electricity"
						onChange={() => {
							setElectricity(!electricity);
						}}
						checked={electricity}
					/>
				</Col>
			</Form.Group>
			<Form.Group as={Row} className="mb-3">
				<Form.Label as="legend" column>
					Internet
				</Form.Label>
				<Col>
					<Switch
						id="internet"
						onChange={() => {
							setInternet(!internet);
						}}
						checked={internet}
					/>
				</Col>
			</Form.Group>
		</>
	);
};
export default AdditionalCosts;
