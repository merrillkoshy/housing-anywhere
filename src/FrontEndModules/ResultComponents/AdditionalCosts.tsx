const AdditionalCosts = ({ costs }: { costs: any }) => {
	return (
		<details>
			<summary>
				<strong id="additionalCosts">Additional Costs</strong>
			</summary>
			<ul>
				<li id="water">Water:{costs?.water ? "✅" : "❌"}</li>
				<li id="gas">Gas:{costs?.gas ? "✅" : "❌"}</li>
				<li id="electricity">Electicity:{costs?.electricity ? "✅" : "❌"}</li>
				<li id="internet">Internet:{costs?.internet ? "✅" : "❌"}</li>
			</ul>
		</details>
	);
};
export default AdditionalCosts;
