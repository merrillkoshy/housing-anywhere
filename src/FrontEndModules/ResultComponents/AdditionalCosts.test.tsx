import React from "react";
import { shallow } from "enzyme";
import AdditionalCosts from "./AdditionalCosts";

describe("Addl Costs module", () => {
	const AdditionalCostsModule = shallow(<AdditionalCosts costs={undefined} />);
	it("renders", () => {
		expect(AdditionalCostsModule).toMatchSnapshot();
	});
	it("finds an li with id water", () => {
		expect(AdditionalCostsModule.find("#water").exists()).toBe(true);
	});
	it("finds an li with id gas", () => {
		expect(AdditionalCostsModule.find("#gas").exists()).toBe(true);
	});
	it("finds an li with id electricity", () => {
		expect(AdditionalCostsModule.find("#electricity").exists()).toBe(true);
	});
	it("finds an li with id internet", () => {
		expect(AdditionalCostsModule.find("#internet").exists()).toBe(true);
	});
});
