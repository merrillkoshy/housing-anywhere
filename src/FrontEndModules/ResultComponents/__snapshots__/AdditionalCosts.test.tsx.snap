// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Addl Costs module renders 1`] = `
<details>
  <summary>
    <strong
      id="additionalCosts"
    >
      Additional Costs
    </strong>
  </summary>
  <ul>
    <li
      id="water"
    >
      Water:
      ❌
    </li>
    <li
      id="gas"
    >
      Gas:
      ❌
    </li>
    <li
      id="electricity"
    >
      Electicity:
      ❌
    </li>
    <li
      id="internet"
    >
      Internet:
      ❌
    </li>
  </ul>
</details>
`;
